lib/webpush/encryption.rb

```diff
@@ -0,0 +1,81 @@
+module Webpush
+  module Encryption
+    extend self
+
+    def encrypt(message, p256dh, auth)
+      group_name = "prime256v1"
+      salt = Random.new.bytes(16)
+
+      server = OpenSSL::PKey::EC.new(group_name)
+      server.generate_key
+      server_public_key_bn = server.public_key.to_bn
+
+      group = OpenSSL::PKey::EC::Group.new(group_name)
+      client_public_key_bn = OpenSSL::BN.new(Base64.urlsafe_decode64(p256dh), 2)
+      client_public_key = OpenSSL::PKey::EC::Point.new(group, client_public_key_bn)
```

Comment:
Creation of the `client_public_key` was simplified by using `Base64.urlsafe_decode64(p256dh), 2` to construct the the `OpenSSL::BN` object.